src/services/task.service.js

```diff
@@ -11,7 +11,7 @@ export const taskService = {
 };
 
 async function query(boardId, groupId) {
-  const board = await boardService.getById(boardId);
+  const board = await boardService.getById(boardId, groupId);
```

Comment:
you dont need to pass groupId to getById. you have the filter for to find the groupId.